### PR TITLE
test timestamp fast-forward attack recovery

### LINKED ISSUE #1713: ngclient: test fast-forward recovery
I think we do not currently test successful rollback (aka fast-forward recovery) for timestamp, snapshot and targets:
  * simulate a fast-forward attack: create repository with valid timestamp/snapshot/targets with version=MAX_VERSION (or something large anyway)
  * update client so it sees the large versions 
  * rotate the affected key (timestamp or snapshot) and set metadata version=1
  * update client, assert that new metadata is accepted by ngclient (even though the version number is lower than the local version)

### PR DESCRIPTION
**Pull request by kairoaraujo**
This test simulates the timestamp fast-forward attack recovery.
It simulates that the timestamp keys were compromised, the attacker
generated a new high version of the timestamp.

The repository generates a new key and rollbacks the timestamp
version to the initial version.

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

Part of #1713 

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: tests/test_updater_top_level_update.py [hunk 1]
```diff
@@ -339,6 +339,35 @@ def test_new_timestamp_expired(self) -> None:
 
         self._assert_files_exist([Root.type])
 
+    def test_new_timestamp_fast_foward_recovery(self) -> None:
+        self._init_updater()
+
+        # health initial version of the timestamp
+        self._run_refresh()
+        self._assert_version_equals(Timestamp.type, 1)
```
Comment (jku) [inline tests/test_updater_top_level_update.py]: this part looks unnecessary?

File: tests/test_updater_top_level_update.py [hunk 2]
```diff
@@ -339,6 +339,35 @@ def test_new_timestamp_expired(self) -> None:
...
+
+        # attacker updates to a higher version and add new package
```
Comment (jku) [inline tests/test_updater_top_level_update.py]: I don't understand "add new package"?

**Review by jku — COMMENTED** [review]
Looks good, thanks!  
* The test seems 100% correct but the first few lines seem unnecessary (for this specific test).
* Could mention the rollback requirement (timestamp key rotation) in a method docstring

Opinions?

File: tests/test_updater_top_level_update.py (hunk 2 above)
Comment (kairoaraujo) [reply tests/test_updater_top_level_update.py]: I added all the scenarios/steps, such as adding packages as target files before/after. I simplified it, and  I left it by mistake.

**Review by jku — APPROVED** [review]
LGTM, the only nitpicks I have are
* the docstring doesn't specify that timestamp rollback requires _timestamp_ key rotation (this is relevant because the snapshot rollback requires rotating both online keys and not everyone knows these differences) ... but the code does make it clear 
* PR is marked as fixing the issue but it only fixes 1/3 (as mentioned snapshot and targets are slightly different cases)

Anyway, I think we can merge this as is, those are really minor. If you do tweak the docstring, feel free to squash to a single commit.

**Pull request merged**